hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestTriggerDBSyncEndpoint.java

```diff
@@ -0,0 +1,143 @@
+  private OzoneConfiguration configuration;
+  private OzoneManagerProtocol ozoneManagerProtocol;
+  private OMMetadataManager omMetadataManager;
+  private ReconOMMetadataManager reconOMMetadataManager;
+
+  @Test
+  public void testTriggerDBSyncEndpointWithOM() throws IOException,
+      AuthenticationException {
+    configuration = new OzoneConfiguration();
+    configuration.set(OZONE_RECON_OM_SNAPSHOT_DB_DIR,
+        temporaryFolder.newFolder().getAbsolutePath());
+    configuration.set(OZONE_RECON_DB_DIR,
+        temporaryFolder.newFolder().getAbsolutePath());
+    configuration.set("ozone.om.address", "localhost:9862");
+
+    ozoneManagerProtocol = mock(OzoneManagerProtocol.class);
+    when(ozoneManagerProtocol.getDBUpdates(any(OzoneManagerProtocolProtos
+        .DBUpdatesRequest.class))).thenReturn(new DBUpdates());
+
+    omMetadataManager =
+        initializeNewOmMetadataManager(temporaryFolder.newFolder());
+    reconOMMetadataManager = getTestReconOmMetadataManager(omMetadataManager,
+            temporaryFolder.newFolder());
+
+    ReconUtils reconUtilsMock = mock(ReconUtils.class);
+    DBCheckpoint checkpoint = omMetadataManager.getStore()
+        .getCheckpoint(true);
+    File tarFile = createTarFile(checkpoint.getCheckpointLocation());
+    HttpURLConnection httpURLConnectionMock = mock(HttpURLConnection.class);
+    try (InputStream inputStream = new FileInputStream(tarFile)) {
+      when(httpURLConnectionMock.getInputStream()).thenReturn(inputStream);
+    }
+    when(reconUtilsMock.makeHttpCall(any(), anyString(), anyBoolean()))
+        .thenReturn(httpURLConnectionMock);
+
+    ReconTaskController reconTaskController = mock(ReconTaskController.class);
+    when(reconTaskController.getReconTaskStatusDao())
+        .thenReturn(mock(ReconTaskStatusDao.class));
+
+    ozoneManagerServiceProvider =
+        new OzoneManagerServiceProviderImpl(configuration,
+            reconOMMetadataManager, reconTaskController, reconUtilsMock,
+            ozoneManagerProtocol);
+    ozoneManagerServiceProvider.start();
+
+    ReconTestInjector reconTestInjector =
+        new ReconTestInjector.Builder(temporaryFolder)
+            .withReconSqlDb()
+            .withReconOm(reconOMMetadataManager)
+            .withOmServiceProvider(ozoneManagerServiceProvider)
+            .addBinding(StorageContainerServiceProvider.class,
+                mock(StorageContainerServiceProviderImpl.class))
+            .addBinding(OzoneStorageContainerManager.class,
+                ReconStorageContainerManagerFacade.class)
+            .withContainerDB()
+            .addBinding(NodeEndpoint.class)
+            .addBinding(MetricsServiceProviderFactory.class)
+            .addBinding(ContainerHealthSchemaManager.class)
+            .addBinding(ReconUtils.class, reconUtilsMock)
+            .addBinding(StorageContainerLocationProtocol.class,
+                mock(StorageContainerLocationProtocol.class))
+            .build();
```

Comment:
It seems this whole chunk should be moved to a `setUp()` method instead. And only leave the relevant part related to the core of the test below here.

Reply:
At first I didn't know if I should put it in a `setUp()` method, as if we later add eg. `/scm` endpoint which will trigger an SCM DB sync, that would need other set up things. But I realised if that happens I will adjust the code accordingly, until then it is much clearer with this approach. Thanks, done in the recent commit.